kratos/processes/parallel_distance_calculator_process.cpp

```diff
@@ -0,0 +1,465 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Riccardo Rossi
+//
+//
+
+// System includes
+#include <string>
+#include <iostream>
+
+// External includes
+
+// Project includes
+#include "containers/model.h"
+#include "includes/model_part.h"
+#include "utilities/geometry_utilities.h"
+#include "utilities/parallel_utilities.h"
+#include "processes/parallel_distance_calculator_process.h"
+
+namespace Kratos
+{
+
+template<unsigned int TDim>
+ParallelDistanceCalculatorProcess<TDim>::ParallelDistanceCalculatorProcess(
+    Model& rModel,
+    Parameters Settings)
+    : ParallelDistanceCalculatorProcess<TDim>(rModel.GetModelPart(Settings["model_part_name"].GetString()),Settings)
+{}
+
+template<unsigned int TDim>
+ParallelDistanceCalculatorProcess<TDim>::ParallelDistanceCalculatorProcess(
+    ModelPart& rModelPart,
+    Parameters Settings)
+    : mrModelPart(rModelPart)
+{
+    Settings.ValidateAndAssignDefaults(this->GetDefaultParameters());
+    mpDistanceVar = &KratosComponents< Variable<double> >::Get(Settings["distance_variable"].GetString());
+    mpAreaVar = &KratosComponents< Variable<double> >::Get(Settings["nodal_area_variable"].GetString());
+
+    mMaxLevels = Settings["max_levels"].GetInt();
+    mMaxDistance = Settings["max_distance"].GetDouble();
+    mCalculateExactDistancesToPlane = Settings["calculate_exact_distances_to_plane"].GetBool();
+}
+
+template<unsigned int TDim>
+ParallelDistanceCalculatorProcess<TDim>::ParallelDistanceCalculatorProcess(
+    ModelPart& rModelPart,
+    const Variable<double>& rDistanceVar,
+    const Variable<double>& rAreaVar,
+    const unsigned int MaxLevels,
+    const double MaxDistance,
+    const bool CalculateExactDistancesToPlane)
+    : mrModelPart(rModelPart),
+      mpDistanceVar(&rDistanceVar),
+      mpAreaVar(&rAreaVar),
+      mMaxLevels(MaxLevels),
+      mMaxDistance(MaxDistance),
+      mCalculateExactDistancesToPlane(CalculateExactDistancesToPlane)
+{}
+
+template<unsigned int TDim>
+const Parameters ParallelDistanceCalculatorProcess<TDim>::GetDefaultParameters() const
+{
+    return Parameters(R"({
+        "model_part_name" : "PLEASE_SPECIFY_MODEL_PART",
+        "distance_variable" : "DISTANCE",
+        "nodal_area_variable" : "NODAL_AREA",
+        "max_levels" : 25,
+        "max_distance" : 1.0,
+        "calculate_exact_distances_to_plane" : false
+    })");
+}
+
+template<unsigned int TDim>
+double ParallelDistanceCalculatorProcess<TDim>::FindMaximumEdgeSize()
```

Comment:
I think this is only used in the edge based solver. I think it should be implemented in there.

Reply:
(feel free to leave it for a future PR)

Reply:
Yep, I'll keep that in mind